rdt/transformers/categorical.py

```diff
@@ -145,9 +151,9 @@ def _get_value(self, category):
 
     def _transform_by_row(self, data):
         """Transform the data row by row."""
-        return data.fillna(np.nan).apply(self._get_value).to_numpy()
+        return data.fillna(np.nan).apply(self._get_value)
```

Comment:
I would try to avoid changing the output type, since the other transform option (`by_category`) does not change it either.
Do things fail if this is not removed?

Reply:
Yep, they crash.

Reply:
I dug a bit more into it. I do not think this change and the one below are necessary if we simply take care of converting the transformed data to `numpy` right before indexing it, in line 197 of the HyperTransformer.

You can see an example of this being done in this other branch: https://github.com/sdv-dev/RDT/blob/7d8c8cfb317fa99c91a8e5170441ff96172f7f43/rdt/hyper_transformer.py#L204

I would rather apply this change to fix the errors and continue to produce a consistent output format (numpy) in all the internal transformer methods.